src/unittests/f32_interval.spec.ts

```diff
@@ -0,0 +1,439 @@
+  expected: [number, number];
+}
+
+g.test('span')
+  .paramsSubcasesOnly<SpanCase>(
+    // prettier-ignore
+    [
+      // Single Intervals
+      { intervals: [[0, 10]], expected: [0, 10]},
+      { intervals: [[0, kValue.f32.positive.max]], expected: [0, kValue.f32.infinity.positive]},
+      { intervals: [[0, kValue.f32.positive.nearest_max]], expected: [0, kValue.f32.positive.nearest_max]},
+      { intervals: [[0, kValue.f32.infinity.positive]], expected: [0, kValue.f32.infinity.positive]},
+      { intervals: [[kValue.f32.negative.min, 0]], expected: [kValue.f32.negative.min, 0]},
+      { intervals: [[kValue.f32.negative.nearest_min, 0]], expected: [kValue.f32.negative.nearest_min, 0]},
+      { intervals: [[kValue.f32.infinity.negative, 0]], expected: [kValue.f32.infinity.negative, 0]},
+
+      // Double Intervals
+      { intervals: [[0, 1], [2, 5]], expected: [0, 5]},
+      { intervals: [[2, 5], [0, 1]], expected: [0, 5]},
+      { intervals: [[0, 2], [1, 5]], expected: [0, 5]},
+      { intervals: [[0, 5], [1, 2]], expected: [0, 5]},
+      { intervals: [[kValue.f32.infinity.negative, 0], [0, kValue.f32.infinity.positive]], expected: [kValue.f32.infinity.negative, kValue.f32.infinity.positive]},
+
+      // Multiple Intervals
+      { intervals: [[0, 1], [2, 3], [4, 5]], expected: [0, 5]},
+      { intervals: [[0, 1], [4, 5], [2, 3]], expected: [0, 5]},
+      { intervals: [[0, 1], [0, 1], [0, 1]], expected: [0, 1]},
+    ]
+  )
+  .fn(t => {
+    const intervals = t.params.intervals.map(arrayToInterval);
+    const expected = arrayToInterval(t.params.expected);
+
+    const got = F32Interval.span(...intervals);
+    t.expect(
+      objectEquals(got, expected),
+      `span({${intervals}}) returned ${got}. Expected ${expected}`
+    );
+  });
+
+interface CorrectlyRoundedCase {
+  value: number;
+  expected: [number, number];
+}
+
+g.test('correctlyRoundedInterval')
+  .paramsSubcasesOnly<CorrectlyRoundedCase>(
+    // prettier-ignore
+    [
+      // Edge Cases
+      { value: kValue.f32.infinity.positive, expected: [kValue.f32.positive.max, Number.POSITIVE_INFINITY] },
+      { value: kValue.f32.infinity.negative, expected: [Number.NEGATIVE_INFINITY, kValue.f32.negative.min] },
+      { value: kValue.f32.positive.max, expected: [kValue.f32.positive.max, kValue.f32.positive.max] },
+      { value: kValue.f32.negative.min, expected: [kValue.f32.negative.min, kValue.f32.negative.min] },
+      { value: kValue.f32.positive.min, expected: [kValue.f32.positive.min, kValue.f32.positive.min] },
+      { value: kValue.f32.negative.max, expected: [kValue.f32.negative.max, kValue.f32.negative.max] },
+
+      // 32-bit subnormals
+      { value: kValue.f32.subnormal.positive.min, expected: [0, kValue.f32.subnormal.positive.min] },
+      { value: kValue.f32.subnormal.positive.max, expected: [0, kValue.f32.subnormal.positive.max] },
+      { value: kValue.f32.subnormal.negative.min, expected: [kValue.f32.subnormal.negative.min, 0] },
```

Comment:
Question for @dneto0 - should this be `-0` ? I've been handling negative zeros in Tint - I'm not sure what the official stance is on them though.

Reply:
Interesting. I know there are tests out there that expect `-0` to be preserved. Now I have to go find them...